app/src/app/constants/layers.ts

```diff
@@ -176,39 +167,45 @@ const addBlockLayers = (
   map?.addSource(BLOCK_SOURCE_ID, blockSource);
   map?.addLayer(
     getBlocksLayerSpecification(mapDocument.parent_layer, BLOCK_LAYER_ID),
-    LABELS_BREAK_LAYER_ID,
+    LABELS_BREAK_LAYER_ID
   );
   map?.addLayer(
     getBlocksHoverLayerSpecification(
       mapDocument.parent_layer,
-      BLOCK_HOVER_LAYER_ID,
+      BLOCK_HOVER_LAYER_ID
     ),
-    LABELS_BREAK_LAYER_ID,
+    LABELS_BREAK_LAYER_ID
   );
   if (mapDocument.child_layer) {
     map?.addLayer(
       getBlocksLayerSpecification(
         mapDocument.child_layer,
-        BLOCK_LAYER_ID_CHILD,
+        BLOCK_LAYER_ID_CHILD
       ),
-      LABELS_BREAK_LAYER_ID,
+      LABELS_BREAK_LAYER_ID
     );
     map?.addLayer(
       getBlocksHoverLayerSpecification(
         mapDocument.child_layer,
-        BLOCK_HOVER_LAYER_ID_CHILD,
+        BLOCK_HOVER_LAYER_ID_CHILD
       ),
-      LABELS_BREAK_LAYER_ID,
+      LABELS_BREAK_LAYER_ID
     );
   }
-  useMapStore.getState().setMapRenderingState("loaded")
+  useMapStore.getState().setMapRenderingState("loaded");
+  useMapStore
+    .getState()
+    .getMapRef()
+    ?.fitBounds(mapDocument.extent as [number, number, number, number], {
```

Comment:
PP: Map bounds could stored in state and have a `useEffect` in the map component to `fitBounds` to decouple from layers add/remove